plugins/inputs/solr/solr.go

```diff
@@ -0,0 +1,395 @@
+		AutocommitMaxDocs        int    `json:"autocommit maxDocs"`
+		AutocommitMaxTime        string `json:"autocommit maxTime"`
+		Autocommits              int    `json:"autocommits"`
+		Commits                  int    `json:"commits"`
+		CumulativeAdds           int    `json:"cumulative_adds"`
+		CumulativeDeletesByID    int    `json:"cumulative_deletesById"`
+		CumulativeDeletesByQuery int    `json:"cumulative_deletesByQuery"`
+		CumulativeErrors         int    `json:"cumulative_errors"`
+		DeletesByID              int    `json:"deletesById"`
+		DeletesByQuery           int    `json:"deletesByQuery"`
+		DocsPending              int    `json:"docsPending"`
+		Errors                   int    `json:"errors"`
+		ExpungeDeletes           int    `json:"expungeDeletes"`
+		Optimizes                int    `json:"optimizes"`
+		Rollbacks                int    `json:"rollbacks"`
+		SoftAutocommits          int    `json:"soft autocommits"`
+	} `json:"stats"`
+}
+
+// Cache is an exported type that
+// contains cache metrics
+type Cache struct {
+	Class string `json:"class"`
+	Stats struct {
+		CumulativeEvictions int     `json:"cumulative_evictions"`
+		CumulativeHitratio  float64 `json:"cumulative_hitratio,string"`
+		CumulativeHits      int     `json:"cumulative_hits"`
+		CumulativeInserts   int     `json:"cumulative_inserts"`
+		CumulativeLookups   int     `json:"cumulative_lookups"`
+		Evictions           int     `json:"evictions"`
+		Hitratio            float64 `json:"hitratio,string"`
+		Hits                int     `json:"hits"`
+		Inserts             int     `json:"inserts"`
+		Lookups             int     `json:"lookups"`
+		Size                int     `json:"size"`
+		WarmupTime          int     `json:"warmupTime"`
+	} `json:"stats"`
+}
+
+// NewSolr return a new instance of Solr
+func NewSolr() *Solr {
+	return &Solr{
+		HTTPTimeout: internal.Duration{Duration: time.Second * 5},
+	}
+}
+
+// SampleConfig returns sample configuration for this plugin.
+func (s *Solr) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns the plugin description.
+func (s *Solr) Description() string {
+	return "Read stats from one or more Solr servers or cores"
+}
+
+// Gather reads the stats from Solr and writes it to the
+// Accumulator.
+func (s *Solr) Gather(acc telegraf.Accumulator) error {
+	if s.client == nil {
+		client, err := s.createHTTPClient()
+
+		if err != nil {
+			return err
+		}
+		s.client = client
+	}
+
+	errChan := errchan.New(len(s.Servers) * len(s.Cores))
+	var wg sync.WaitGroup
+
+	for _, serv := range s.Servers {
+		for _, core := range s.Cores {
+			wg.Add(1)
+			go func(serv string, core string, acc telegraf.Accumulator) {
+				defer wg.Done()
+				if err := s.gatherCoreStats(fmt.Sprintf("%s/solr/%s"+mbeansPath, serv, core), core, acc); err != nil {
+					errChan.C <- err
+					return
+				}
+			}(serv, core, acc)
+		}
+	}
+	wg.Wait()
+	return errChan.Error()
+}
+
+func (s *Solr) createHTTPClient() (*http.Client, error) {
```

Comment:
This function doesn't look like it ever returns an error, so it probably doesn't need to have one defined in the return signature.